### PR TITLE
Add discount ribbon

### PR DESCRIPTION
**Pull request by ahmardi**

### CODE REVIEW

File: view/frontend/web/js/amwal-magento-react-button/src/AmwalMagentoReactButton.tsx [hunk 1]
```diff
@@ -351,7 +357,8 @@ const AmwalMagentoReactButton = ({
         postcodeOptionalCountries={JSON.stringify(config.post_code_optional_countries) as any}
         initialFirstName={config.initial_first_name}
         initialLastName={config.initial_last_name}
-        installmentOptionsUrl={config.installment_options_url}
+        showDiscountRibbon={config.show_discount_ribbon}
+        // installmentOptionsUrl={config.installment_options_url}
```
Comment (sameh-amwal) [inline view/frontend/web/js/amwal-magento-react-button/src/AmwalMagentoReactButton.tsx:361]:
```suggestion
        installmentOptionsUrl={config.installment_options_url}
```

File: view/frontend/templates/express/checkout-button.phtml [hunk 2]
```diff
@@ -22,6 +22,7 @@ $formSelector = $viewModel->getFormSelector($this->getFormSelector(), $block->ge
          data-trigger-context="<?= $escaper->escapeHtmlAttr($triggerContext) ?>"
          data-locale="<?= $escaper->escapeHtmlAttr($viewModel->getLocale()) ?>"
          data-scope-code="<?= $escaper->escapeHtmlAttr($viewModel->getStoreCode()) ?>"
+         data-product-id="<?= $escaper->escapeHtmlAttr($viewModel->getProductId()) ?>"
```
Comment (sameh-amwal) [inline view/frontend/templates/express/checkout-button.phtml:25]: use `$block->getProduct()` to get the product id instead

**Review by sameh-amwal — APPROVED** [review]

File: Api/Data/AmwalButtonConfigInterface.php [hunk 3]
```diff
@@ -335,4 +335,27 @@ public function getCartId(): ?string;
      */
     public function setCartId(?string $cartId): AmwalButtonConfigInterface;
 
+
```
Comment (Skullsneeze) [inline Api/Data/AmwalButtonConfigInterface.php:338]:
```suggestion
```

File: Model/Button/GetCartButtonConfig.php [hunk 4]
```diff
@@ -53,32 +56,76 @@ public function execute(
             $buttonConfig->setCartId($cartId);
         }
         $this->addGenericButtonConfig($buttonConfig, $refIdData, $quote, $customerSession, $initialAddress);
+        if ($triggerContext === 'regular-checkout') {
```
Comment (Skullsneeze) [inline Model/Button/GetCartButtonConfig.php:59]: Let's maybe introduce a constant here for maintainability (defined together with the other constants for trigger context):
```suggestion
        if ($triggerContext ===  ExpressCheckoutButton::TRIGGER_CONTEXT_REGULAR_CHECKOUT) {
```

File: Model/Button/GetCartButtonConfig.php [hunk 5]
```diff
@@ -53,32 +56,76 @@ public function execute(
...
      * @throws NoSuchEntityException
      */
-    public function getAmount($quote): float
+    public function getAmount($quote, AmwalButtonConfigInterface $buttonConfig, $productId = null): float
     {
+        if ($buttonConfig->getShowDiscountRibbon()) {
+            if ($productId) {
+                $product = $this->productRepository->getById($productId);
+                return (float)$product->getPriceInfo()->getPrice('regular_price')->getAmount()->getValue();
+            }else {
```
Comment (Skullsneeze) [inline Model/Button/GetCartButtonConfig.php:90]:
```suggestion
            } else {
```

File: Model/Button/GetCartButtonConfig.php [hunk 6]
```diff
@@ -53,32 +56,76 @@ public function execute(
...
+     */
+    public function getDiscountAmount($quote, AmwalButtonConfigInterface $buttonConfig, $productId = null): float
+    {
+        $discountAmount = 0;
+        if ($buttonConfig->getShowDiscountRibbon()) {
+            if ($productId) {
+                $product = $this->productRepository->getById($productId);
+                $priceInfo = $product->getPriceInfo();
+                $discountAmount += $priceInfo->getPrice('regular_price')->getAmount()->getValue() - $priceInfo->getPrice('final_price')->getAmount()->getValue();
+            }else {
```
Comment (Skullsneeze) [inline Model/Button/GetCartButtonConfig.php:118]:
```suggestion
            } else {
```

**Review by Skullsneeze — CHANGES_REQUESTED** [review]
Almost there I think :)

**Pull request merged**